packages/lesswrong/server/exportUserData.ts

```diff
@@ -0,0 +1,85 @@
+import Messages from "../lib/collections/messages/collection";
+import { ModeratorActions } from "../lib/collections/moderatorActions";
+import Notifications from "../lib/collections/notifications/collection";
+import { PetrovDayLaunchs } from "../lib/collections/petrovDayLaunchs";
+import Posts from "../lib/collections/posts/collection";
+import ReadStatuses from "../lib/collections/readStatus/collection";
+import Reports from "../lib/collections/reports/collection";
+import Revisions from "../lib/collections/revisions/collection";
+import RSSFeeds from "../lib/collections/rssfeeds/collection";
+import Sequences from "../lib/collections/sequences/collection";
+import { Subscriptions } from "../lib/collections/subscriptions";
+import TagRels from "../lib/collections/tagRels/collection";
+import Tags from "../lib/collections/tags/collection";
+import UserMostValuablePosts from "../lib/collections/userMostValuablePosts/collection";
+import Users from "../lib/collections/users/collection";
+import UserTagRels from "../lib/collections/userTagRels/collection";
+import { Votes } from "../lib/collections/votes";
+import { Globals } from "./vulcan-lib";
+import { writeFile } from "fs/promises";
+
+export const exportUserData = async (
+  {_id, slug}: {_id?: string, slug?: string},
+  outfile?: string,
+) => {
+  if (!_id && !slug) {
+    throw new Error("Must specify either a user ID or slug");
+  }
+
+  const user = await Users.findOne({_id, slug});
+  if (!user) {
+    throw new Error("User not found");
+  }
+
+  const userId = user._id;
+
+  const entries: [string, any][] = [
+    ["user", user],
+    ["bans", Bans.find({userId}).fetch()],
+    ["clientIds", ClientIds.find({userIds: userId}).fetch()],
+    ["collections", Collections.find({userId}).fetch()],
+    ["comments", Comments.find({userId}).fetch()],
+    ["conversations", Conversations.find({participantIds: userId}).fetch()],
+    ["emailTokens", EmailTokens.find({userId}).fetch()],
+    ["localGroups", Localgroups.find({organizerIds: userId}).fetch()],
+    ["messages", Messages.find({userId}).fetch()],
+    ["moderatorActions", ModeratorActions.find({userId}).fetch()],
+    ["notifications", Notifications.find({userId}).fetch()],
+    ["pretrovDayLaunches", PetrovDayLaunchs.find({userId}).fetch()],
+    ["posts", Posts.find({userId}).fetch()],
+    ["rssFeeds", RSSFeeds.find({userId}).fetch()],
+    ["readStatuses", ReadStatuses.find({userId}).fetch()],
+    ["reports", Reports.find({userId}).fetch()],
+    ["revisions", Revisions.find({userId}).fetch()],
+    ["sequences", Sequences.find({userId}).fetch()],
+    ["subscriptions", Subscriptions.find({userId}).fetch()],
+    ["tagRels", TagRels.find({userId}).fetch()],
+    ["tags", Tags.find({userId}).fetch()],
+    ["userMostValuablePosts", UserMostValuablePosts.find({userId}).fetch()],
+    ["userTagRels", UserTagRels.find({userId}).fetch()],
+    ["votes", Votes.find({userId}).fetch()],
+  ];
```

Comment:
Can we pass this whole list through an access filter, and only give them what they're allowed access to? (Sorry! It does feel like it prevents a whole class of issues though.) That does remove some collections.